src/components/TeacherDonation/DonationLayout.js

```diff
@@ -1,11 +1,56 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Image, Menu, Icon } from 'semantic-ui-react';
-import { Navigate, Link, Outlet } from 'react-router-dom';
+import { Link, Outlet, useParams } from 'react-router-dom';
+import SupplyService from '../../services/supply.service';
+import logo from '../../media/logo.png';
 
 
 function DonationLayout() {
 
-    const [teacher, setTeacher] = useState('John Doe');
+    const { teacherId } = useParams();
+
+    console.log("Teacher id is: " + teacherId);
+
```

Comment:
I just added school and message to the Teacher model so you can asign these now :)